spanner/api/Program.cs

```diff
@@ -793,10 +746,40 @@ AlbumTitle   STRING(MAX)
         }
 
         public static object QueryDataWithIndex(string projectId,
-string instanceId, string databaseId)
+            string instanceId, string databaseId,
+            string startTitle, string endTitle)
         {
-            var response = Task.Run(async () => await QueryDataWithIndexAsync(
-                projectId, instanceId, databaseId));
+            var response = new Task(() => { });
+            // Call method QueryDataWithIndexAsync() based on whether 
+            // user provided startTitle and endTitle values are empty or null.
+            if (string.IsNullOrEmpty(startTitle))
+            {
+                // startTitle not provided, exclude startTitle and endTitle from 
+                // method call to use default values for both.
+                response = Task.Run(async () => await QueryDataWithIndexAsync(
+                        projectId, instanceId, databaseId));
+            }
+            else if (!string.IsNullOrEmpty(startTitle)
+                && string.IsNullOrEmpty(endTitle))
+            {
+                // Only startTitle provided, show message that user must 
+                // provide both startTitle and endTitle or exclude them 
+                // from their command.
+                Console.WriteLine("Please provide values for both the start "
+                    + "of the title index and the end of the title index.");
+                Console.WriteLine("Or you can exclude both to run the "
+                    + "query with default values.");
+                return 0;
+            }
+            else if (!string.IsNullOrEmpty(startTitle)
+                && !string.IsNullOrEmpty(endTitle))
+            {
+                // Both startTitle and endTitle provided, include them both in
+                // the method call to QueryDataWithIndexAsync.
+                response = Task.Run(async () => await QueryDataWithIndexAsync(

```

Comment:
this can just be  response = QueryDataWithIndexAsync(....);

shouldnt need task.run